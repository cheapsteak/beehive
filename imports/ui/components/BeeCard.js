// Framework
import React from "react";

// Components
import {
  Card,
  CardImg,
  CardBlock,
  CardTitle,
  CardSubtitle,
  Button
} from "reactstrap";

/**
 * A bee reporting card.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const BeeCard = ({ bee: { _id, type, name }, removeBee }) => {
  const remove = () => removeBee(_id);

  return (
    <Card className="bee-card">
      <CardImg
        width="100%"
        src={`./images/bees/${type.toLowerCase()}.jpg`}
        alt={`${type} Bee`}
        top
      />
      <CardBlock>
        <CardTitle>{name}</CardTitle>
        <CardSubtitle>{type}</CardSubtitle>
        <Button color="danger" onClick={remove}>
          Delete
        </Button>
      </CardBlock>
    </Card>
  );
};

export default BeeCard;
