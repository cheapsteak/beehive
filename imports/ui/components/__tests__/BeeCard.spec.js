import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';

import ReactTestRenderer from 'react-test-renderer';
import ReactShallowRenderer from 'react-test-renderer/shallow';

import BeeCard from '../BeeCard';

import {
  Card,
  CardImg,
  CardBlock,
  CardTitle,
  CardSubtitle,
  Button
} from 'reactstrap';

describe('BeeCard', () => {
  let instance;
  let cardImg, cardBlock;
  let cardTitle, cardSubtitle, deleteButton;
  const renderer = new ReactShallowRenderer();

  let bee = { _id: 'testID', name: 'test bee', type: 'test type' };
  let removeBee = jest.fn();

  describe('BeeCard SnapShot', () => {
    it('renders BeeCard correctly', () => {
      const tree = ReactTestRenderer.create(
        <BeeCard bee={bee} removeBee={removeBee} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when bee is a worker', () => {
    beforeEach(() => {
      renderer.render(<BeeCard bee={bee} removeBee={removeBee} />);
      instance = renderer.getRenderOutput();

      [cardImg, cardBlock] = instance.props.children;
      [cardTitle, cardSubtitle, deleteButton] = cardBlock.props.children;
    });

    it('checks rendering of CardImg', () => {
      const {
        type,
        props: { src, alt }
      } = cardImg;
      expect(type).toBe(CardImg);
      expect(src).toBe(`./images/bees/${bee.type.toLowerCase()}.jpg`);
      expect(alt).toBe(`${bee.type} Bee`);
    });

    it('checks rendering of CardTitle', () => {
      const {
        type,
        props: { children }
      } = cardTitle;
      expect(type).toBe(CardTitle);
      expect(children).toBe(bee.name);
    });

    it('checks rendering of CardSubtitle', () => {
      const {
        type,
        props: { children }
      } = cardSubtitle;
      expect(type).toBe(CardSubtitle);
      expect(children).toBe(bee.type);
    });

    it('checks to see if the deleteButton calls removeBee', () => {
      deleteButton.props.onClick();
      expect(removeBee).toHaveBeenCalled();
    });
  });
});
