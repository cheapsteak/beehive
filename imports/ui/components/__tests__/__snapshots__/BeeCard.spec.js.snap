// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeeCard BeeCard SnapShot renders BeeCard correctly 1`] = `
<div
  className="bee-card card"
>
  <img
    alt="test type Bee"
    className="card-img-top"
    src="./images/bees/test type.jpg"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      test bee
    </h5>
    <h6
      className="card-subtitle"
    >
      test type
    </h6>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
